### ISSUE TITLE
no-reserved-keywords shouldn't ban "type"

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/tslint-microsoft-contrib#403]
https://stackoverflow.com/questions/33446498/is-type-a-keyword-in-javascript

It's not part of the ES6 spec. Just a TS/Flow construct.

**Comment by JoshuaKGoldberg** [comment]
Huh, looks like this is superseded by tslint's `variable-name` rule. Never mind!

**Comment by lijunle** [comment]
I suppose this is still an issue of TSLint. Isn't it?

**Comment by JoshuaKGoldberg** [comment]
Fortunately not.

https://github.com/palantir/tslint/blob/235561e411c58ecbd4d96844fae28e998f0ccbda/src/rules/variableNameRule.ts#L26

```typescript
const BANNED_KEYWORDS = ["any", "Number", "number", "String", "string", "Boolean", "boolean", "Undefined", "undefined"];
```

**Comment by astorije** [comment]
Hey @JoshuaKGoldberg!

[This](https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/src/noReservedKeywordsRule.ts#L87) should still be removed, here, no?

The use cases we're seeing, at least at the moment, are interface definitions that contain `type: string`. One use case is a return type from GraphQL (meaning, the backend API returns a `type` field and that's what it is), another one is we're exposing an text input `type` (`password`, etc.) from an underlying library.

**Comment by JoshuaKGoldberg** [comment]
@astorije agreed, `type` should be removed. I think the root problem is that this rule should be marked deprecated since `variable-name` overrides it, but hasn't yet been. Filed #652. Thanks for the ping!

**Comment by birkir** [comment]
Should be removed:

```ts
interface MyGraphQLType {
  type: GraphQLOutputType;
}
```
